src/coreclr/binder/inc/assemblyidentitycache.hpp

```diff
@@ -1,114 +0,0 @@
-// The .NET Foundation licenses this file to you under the MIT license.
-// ============================================================
-//
-// AssemblyIdentityCache.hpp
-//
-
-
-//
-// Defines the AssemblyIdentityCache class and its helpers
-//
-// ============================================================
-
-#ifndef __BINDER__ASSEMBLY_IDENTITY_CACHE_HPP__
-#define __BINDER__ASSEMBLY_IDENTITY_CACHE_HPP__
-
-#include "bindertypes.hpp"
-#include "assemblyidentity.hpp"
-#include "sstring.h"
-#include "shash.h"
-
-namespace BINDER_SPACE
-{
-    class AssemblyIdentityCacheEntry
-    {
-    public:
-        inline AssemblyIdentityCacheEntry()
-        {
-            m_szTextualIdentity = NULL;
-            m_pAssemblyIdentity = NULL;
-        }
-        inline ~AssemblyIdentityCacheEntry()
-        {
-            SAFE_DELETE_ARRAY(m_szTextualIdentity);
-            SAFE_DELETE(m_pAssemblyIdentity);
-        }
-
-        // Getters/Setters
-        inline LPCSTR GetTextualIdentity()
-        {
-            return m_szTextualIdentity;
-        }
-        inline void SetTextualIdentity(LPCSTR szTextualIdentity)
-        {
-            size_t len = strlen(szTextualIdentity) + 1;
-
-            m_szTextualIdentity = new char[len];
-            strcpy_s((LPSTR) m_szTextualIdentity, len, szTextualIdentity);
-        }
-        inline AssemblyIdentityUTF8 *GetAssemblyIdentity()
-        {
-            return m_pAssemblyIdentity;
-        }
-        inline void SetAssemblyIdentity(AssemblyIdentityUTF8 *pAssemblyIdentity)
-        {
-            m_pAssemblyIdentity = pAssemblyIdentity;
-        }
-
-    protected:
-        LPCSTR                m_szTextualIdentity;
-        AssemblyIdentityUTF8 *m_pAssemblyIdentity;
-    };
-
-    class AssemblyIdentityHashTraits : public DefaultSHashTraits<AssemblyIdentityCacheEntry *>
-    {
-    public:
-        typedef LPCSTR key_t;
-
-        static key_t GetKey(element_t pAssemblyIdentityCacheEntry)
-        {
-            return pAssemblyIdentityCacheEntry->GetTextualIdentity();
-        }
-        static BOOL Equals(key_t textualIdentity1, key_t textualIdentity2)
-        {
-            if ((textualIdentity1 == NULL) && (textualIdentity2 == NULL))
-                return TRUE;
-            if ((textualIdentity1 == NULL) || (textualIdentity2 == NULL))
-                return FALSE;
-
-            return (strcmp(textualIdentity1, textualIdentity2) == 0);
-        }
-        static count_t Hash(key_t textualIdentity)
-        {
-            if (textualIdentity == NULL)
-                return 0;
-            else
-                return HashStringA(textualIdentity);
-        }
-        static element_t Null()
-        {
-            return NULL;
-        }
-        static bool IsNull(const element_t &assemblyIdentityCacheEntry)
-        {
-            return (assemblyIdentityCacheEntry == NULL);
-        }
-
-    };
-
-    class AssemblyIdentityCache : protected SHash<AssemblyIdentityHashTraits>
```

Comment:
I wonder what we used to cache here.

Reply:
I looks like it was caching the results of parsing + some strings in UTF8 form. I do not see if the cache had any size limit, so the likely expectations were that we parse just a few distinct strings, but frequently.

We can introduce some form of caching on the managed side, but may need to see if there is a need for that.

Reply:
This cache has no limit. It was caching everything that was ever parsed in the process. You can create test cases (e.g. with unloadable assemblies) where this cache grows indefinitely.

I believe that this cache was a left-over from fusion where the assembly name parsing implementation was a very complex and slow algorithm and there was a lot of redundant assembly name parsing. I do not think we need it anymore.

The most frequently exercises assembly name parsing paths stopped using the cache in your earlier PR. I believe that you have done some performance testing to show that the managed parser is as fast as the unmanaged one.

Reply:
Yes. I think the parsing has acceptable performance and allocation profile vs. how often we need to parse. 

If we get more evidence that caching is needed, we can think of something, but we should not introduce caching just because the native implementation had it.